modules/monitoring/library/Monitoring/Web/Controller/MonitoredObjectController.php

```diff
@@ -273,4 +283,107 @@ protected function setupQuickActionForms()
                 ->handleRequest();
         }
     }
+
+    /**
+     * Get the monitored object's check command, i.e. the command executed during the last check
+     *
+     * Only available if there's at least one Icinga 2 API command transport
+     *
+     * @return string|null
+     */
+    protected function getCheckCommand()
+    {
+        try {
+            $api = Icinga2Api::fromTransport();
+        } catch (ConfigurationError $e) {
+            return null;
+        }
+
+        if ($this->object instanceof Service) {
+            $objects = $api->objects(
+                'services',
+                Filter::where('host.name', $this->object->getHost()->getName())->andFilter(
+                    Filter::where('service.name', $this->object->getName())
+                )
+            );
+        } else {
+            $objects = $api->objects('hosts', Filter::where('host.name', $this->object->getName()));
+        }
+
+        if (! isset($objects['results'][0]['attrs']['last_check_result']['command'])) {
+            return null;
+        }
+
+        $result = array();
+
+        $checkCommands = $api->objects('checkcommands', Filter::where(
+            'checkcommand.name',
+            $this->object instanceof Service
+                ? $this->object->service_check_command
+                : $this->object->host_check_command
+        ));
+
+        if (isset($checkCommands['results'][0]['attrs'])) {
+            $attrs = $checkCommands['results'][0]['attrs'];
+
+            if (! isset($attrs['vars'])) {
+                $attrs['vars'] = array();
+            }
+
+            if (isset($attrs['env'])) {
+                foreach ($attrs['env'] as $key => $value) {
+                    $result[] = "$key={$this->escapeShellArg($this->resolveCommandMacros($value, $this->object, $attrs['vars']))}";
+                }
+            }
+        }
+
+        foreach ($objects['results'][0]['attrs']['last_check_result']['command'] as $arg) {
```

Comment:
Small not here: Checks can be pending, e.g. a fresh host/service object after a new config deployment. Then `last_check_result` returns null, and the `command` key lookup will break. You'll need another sanity check here if `last_check_result` actually is a dictionary.

Reply:
No, it won't break as I've done an isset() before.